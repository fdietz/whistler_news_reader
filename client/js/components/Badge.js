import React, { PropTypes } from 'react';
import classNames from 'classnames';

const Badge = ({ count, className }) => {
  const cls = classNames('badge', className);

  return (
    <div className={cls}>{count > 99 ? '99+' : count}</div>
  );
};

Badge.propTypes = {
  count: PropTypes.number,
  className: PropTypes.string,
};

export default Badge;
