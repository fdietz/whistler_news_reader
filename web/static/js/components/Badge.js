import React, { PropTypes } from "react";

const Badge = ({count, className}) => {
  const cls = `badge ${className}`;

  return (
    <div className={cls}>{count > 99 ? "99+" : count}</div>
  );
};

Badge.propTypes = {
  count: PropTypes.number,
  className: PropTypes.string
};

export default Badge;
