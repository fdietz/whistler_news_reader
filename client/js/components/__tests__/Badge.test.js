import renderer from 'react-test-renderer';
import React from 'react';
import Badge from '../Badge';

describe('Badge', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Badge count={1} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders count 99+ if large', () => {
    const tree = renderer.create(<Badge count={100} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
