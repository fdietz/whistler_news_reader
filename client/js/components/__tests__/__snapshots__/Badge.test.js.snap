exports[`Badge renders correctly 1`] = `
<div
  className="badge">
  1
</div>
`;

exports[`Badge renders count 99+ if large 1`] = `
<div
  className="badge">
  99+
</div>
`;
